File: src/dataset/cpp/grade-school_exercism-chrisnp/openAI/ChatGPT4_grade_school_v3.cpp
#include "grade_school.h"
#include <algorithm>
#include <stdexcept>
#include <unordered_map>

using namespace std;

grade_school::Roster grade_school::school::roster() const {
    return school_;
}

void grade_school::school::add(string student, int grade) {
    auto &students = school_[grade];
    if (find(students.begin(), students.end(), student) == students.end()) {
        students.push_back(move(student));
        inplace_merge(students.begin(), students.end() - 1, students.end());
    }
}

vector<string> grade_school::school::grade(int grade) const {
    auto it = school_.find(grade);
    return it != school_.end() ? it->second : vector<string>{};
}

File: src/dataset/javascript/grade-school_exercism-ThomasZumsteg/openAI/ChatGPT4_grade-school_v3.js
class School {
  constructor() {
    this.class = new Map();
  }

  roster() {
    const result = {};
    for (const [grade, students] of this.class.entries()) {
      result[grade] = [...students];
    }
    return result;
  }

  grade(grade) {
    return this.class.get(grade) ? [...this.class.get(grade)] : [];
  }

  add(student, grade) {
    if (!this.class.has(grade)) {
      this.class.set(grade, new Set());
    }
    this.class.get(grade).add(student);
    this.class.set(grade, new Set([...this.class.get(grade)].sort()));
  }
}

module.exports = School;